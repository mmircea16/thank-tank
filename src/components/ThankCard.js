import React, { Component } from 'react';

import ThankText from './ThankText'

export default class ThankCard extends Component{
    constructor(props){
        super(props);
        this.state = { ...props, autoScroll: false};
        this.shouldAutoScroll = this.shouldAutoScroll.bind(this);
    }

    shouldAutoScroll() {
        var result = (this.props.flipped === 'flipped' && this.props.className.includes('back')) ||
                     (this.props.flipped !== 'flipped' && this.props.className.includes('front'));

        if(result !== this.state.autoScroll){
            this.setState({autoScroll: result});
        }
    }

    componentDidUpdate(){
        if(!this.state.autoScroll)
        {
            setTimeout(this.shouldAutoScroll, 4000); 
        } else{
            this.shouldAutoScroll();
        }       
    }

    componentDidMount(){
        this.shouldAutoScroll();
    }

    render(){     
        //this.shouldAutoScroll();  
        var row = this.props.row;
        //var thankedBy = row[4] === 'Yes'? 'Anonymous' : row[1];
        return (<article className={'thankCard ' + this.props.className}>
            <header><strong>Thank you <span className="name">{row.name}!</span></strong></header>
            <ThankText message={row.message} autoScrollEnabled={this.state.autoScroll}/>
            <footer>from {row.from}</footer>
        </article>);
    }
}