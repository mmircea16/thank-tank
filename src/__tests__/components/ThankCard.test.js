'use strict';

import React from 'react';
import ThankCard from '../../components/ThankCard';
import renderer from 'react-test-renderer';

describe("ThankCard", () => {
  it('renders correctly if secret is yes', () => {
    const tree = renderer
      .create(<ThankCard className='testClass' row={{from: 'Anonymous', name: 'Thanking to', message:'message'}} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly if secret is no', () => {
    const tree = renderer
      .create(<ThankCard className='testClass' row={{from: 'email', name: 'Thanking to', message:'message'}} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});