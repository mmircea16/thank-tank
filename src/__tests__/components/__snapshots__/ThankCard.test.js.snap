// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThankCard renders correctly if secret is no 1`] = `
<article
  className="thankCard testClass"
>
  <header>
    <strong>
      Thank you 
      <span
        className="name"
      >
        Thanking to
        !
      </span>
    </strong>
  </header>
  <div
    className="autoscrollcomponent"
  >
    <section>
      message
    </section>
  </div>
  <footer>
    from 
    email
  </footer>
</article>
`;

exports[`ThankCard renders correctly if secret is yes 1`] = `
<article
  className="thankCard testClass"
>
  <header>
    <strong>
      Thank you 
      <span
        className="name"
      >
        Thanking to
        !
      </span>
    </strong>
  </header>
  <div
    className="autoscrollcomponent"
  >
    <section>
      message
    </section>
  </div>
  <footer>
    from 
    Anonymous
  </footer>
</article>
`;
